src/main/java/com/hierynomus/smbj/transport/tcp/DirectTcpTransport.java

```diff
@@ -15,31 +15,131 @@
  */
 package com.hierynomus.smbj.transport.tcp;
 
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.concurrent.locks.ReentrantLock;
+
+import javax.net.SocketFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.hierynomus.protocol.Packet;
 import com.hierynomus.protocol.commons.buffer.Buffer;
-import com.hierynomus.smbj.transport.BaseTransport;
-
-import java.io.IOException;
+import com.hierynomus.protocol.commons.socket.ProxySocketFactory;
+import com.hierynomus.smbj.transport.PacketHandlers;
+import com.hierynomus.smbj.transport.PacketReader;
+import com.hierynomus.smbj.transport.TransportException;
+import com.hierynomus.smbj.transport.TransportLayer;
 
 /**
- * A transport layer to do SMB2 over Direct TCP/IP.
+ * A transport layer over Direct TCP/IP.
  */
-public class DirectTcpTransport<P extends Packet<P, ?>> extends BaseTransport<P> {
+public class DirectTcpTransport<P extends Packet<P,?>> implements TransportLayer<P> {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final PacketHandlers<P> handlers;
+
+    private final ReentrantLock writeLock = new ReentrantLock();
+
+    private SocketFactory socketFactory = new ProxySocketFactory();
+    private int soTimeout;
+
+    private Socket socket;
+    private BufferedOutputStream output;
+    private PacketReader<P> packetReaderThread;
+
+    private static final int INITIAL_BUFFER_SIZE = 9000;
+
+    public DirectTcpTransport(SocketFactory socketFactory, int soTimeout, PacketHandlers<P> handlers) {
+        this.soTimeout = soTimeout;
+        this.socketFactory = socketFactory;
+        this.handlers = handlers;
+    }
+
+    @Override
+    public void write(P packet) throws TransportException {
+        logger.trace("Acquiring write lock to send packet << {} >>", packet);
+        writeLock.lock();
+        try {
+            try {
+                logger.debug("Writing packet {}", packet);
+                Buffer<?> packetData = handlers.getSerializer().write(packet);
+                writePacketSize(packetData.available());
```

Comment:
Rename to `writeDirectTcpPacketHeader`. This is not only the size, but also a leading `0` byte. See the [MS-SMB2] spec section 2.1